src/coreclr/inc/CrstTypes.def

```diff
@@ -87,6 +87,7 @@ End
 
 Crst AssemblyLoader
     AcquiredBefore DeadlockDetection UniqueStack DebuggerMutex
+                   ThreadStore GlobalStrLiteralMap LoaderAllocatorReferences
 End
```

Comment:
**Design question:**

Managed code are starting to be called under AssemblyLoader lock. Currently only DeclareDependencyOnMvid for R2R path, but I assume that after more components converted to managed, there will be more chance to invoke managed code.

Currently there is no dependency cycle on this. But I want to ensure this is correct in architecture and won't block future changes.